### PR TITLE
Load the template config YAML file from the parent directory

### PR DESCRIPTION
**Pull request by joshmcarthur**
The README describes placing the config YAML file in your working directory, and running `rails new ....` to allow the template to pick up the config file. 

The problem with this, as far as I can tell, is that `template.rb` is evaluated in the context of the directory `rails new` has just created, which does not contain the config file. I think the solution is to traverse up a level and load the config file from there. This won't work if the user has provided an absolute path, but in that case `ENV["CONFIG_PATH"]` can be provided to override it.

As an alternative workaround, a user could create the directory for their app, place the config file inside it, and then run `rails new` (without this patch)

### CODE REVIEW

**Review by G-Rath — APPROVED** [review]
hmm might be nice to have a new test just to be safe... 😜

**Review by halles — APPROVED** [review]

**Review by eoinkelly — APPROVED** [review]

**Comment by joshmcarthur** [comment]
@G-Rath you kid, but I do think it's useful for us to generate an app with the default config, since that's the primary way we suggest people generate a new app. I've adjusted the build script to default to using the default config in the absence of an override.

@eoinkelly you might be interested in re-reviewing this as well since I've messed with your script.

File: ci/bin/build-and-test [hunk 1]
```diff
@@ -7,7 +7,19 @@ template_path = File.join(root_path, "template.rb")
 builds_path = File.join(root_path, "tmp/builds")
 app_name = ENV.fetch("APP_NAME", "app")
 app_path = File.join(builds_path, app_name)
-config_path = File.join(root_path, ENV.fetch("CONFIG_PATH", "ci/configs/basic.yml"))
+
+config_path = if !ENV.key?("CONFIG_PATH") || ENV["CONFIG_PATH"] == ""
```
Comment (G-Rath) [inline ci/bin/build-and-test:11]: can't we do

```suggestion
config_path = if ENV.fetch("CONFIG_PATH", "").blank?
```

Comment (joshmcarthur) [reply ci/bin/build-and-test:11]: @G-Rath `#blank?` comes from ActiveSupport, which is not loaded here. I will adjust your suggestion though to use `fetch`

**Review by robotdana — APPROVED** [review]

**Review by eoinkelly — APPROVED** [review]

File: ci/bin/build-and-test [hunk 2]
```diff
@@ -7,7 +7,19 @@ template_path = File.join(root_path, "template.rb")
...
 app_path = File.join(builds_path, app_name)
-config_path = File.join(root_path, ENV.fetch("CONFIG_PATH", "ci/configs/basic.yml"))
+
+config_path = if ENV.fetch("CONFIG_PATH", "") == ""
+                FileUtils.cp(
+                  File.join(root_path, "ackama_rails_template.config.yml"),
+                  File.join(builds_path, "ackama_rails_template.config.yml")
+                )
+
+                nil
```
Comment (eoinkelly) [inline ci/bin/build-and-test:17]: Having two copies of the `ackama_rails_template.config.yml` might be a _bit_ confusing for devs working on this project. Could we avoid this by setting `config_path` to the top level file if no `CONFIG_PATH` env var provided?

```suggestion
                File.join(root_path, "ackama_rails_template.config.yml"),
```

**Comment by lukeify** [comment]
Discussed at our Ruby catchup and have agreed it will be merged.

**Pull request merged**